lang/cs/Org.Apache.REEF.Common/metrics/Api/AbstractMetric.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System.Collections.Generic;
+
+namespace Org.Apache.REEF.Common.metrics.Api
+{
+    /// <summary>
+    /// Abstract immutable metric class. All metrics put in the record by the MetricsSource are 
+    /// kept as <see cref="IEnumerable{T}" />.
+    /// </summary>
+    public abstract class AbstractMetric : IMetricsInfo
+    {
+        private readonly IMetricsInfo _info;
+
+        private readonly long _value;
+
+        /// <summary>
+        /// Protected constructor of abstract metric.
+        /// </summary>
+        /// <param name="info">Meta-data for the metric</param>
+        /// <param name="value">The value of the metric. It is assumed that user 
+        /// or application has converted float/double values to long via some logic.</param>
```

Comment:
>or application has converted float/double values to long via some logic [](start = 12, length = 71)

why not to make it generic class? doesn't look like there is any dependency on long type